File: functional/repository/repository.go
package repository

import (
	"github.com/yoshiken0927/paradigms/functional/model"
)

func FindEmployee(name string) model.Employee {
	return model.Employee{
		Name:    name,
		Age:     29,
		Version: 0,
	}
}

func UpdateEmployee(e model.Employee) model.Employee {
	return model.Employee{
		Name:    e.Name,
		Age:     e.Age,
		Version: e.Version + 1,
	}

}

File: functional/repository/Repository.hs
module Repository
    ( findEmployee,
      updateEmployee,
    ) where

import Model

findEmployee :: String -> Employee
findEmployee name = Employee {
  name = name,
  age = 30,
  version = 0
}


updateEmployee :: Employee -> Employee
updateEmployee e = Employee {
  name = name(e),
  age = age(e),
  version = version(e) + 1
}


